mlrun/api/api/endpoints/feature_store.py

```diff
@@ -397,6 +398,79 @@ def ingest_feature_set(
     )
 
 
+@router.post(
+    "/projects/{project}/feature-sets/{name}/references/{reference}/publish",
+    response_model=schemas.FeatureSetPublishOutput,
+)
```

Comment:
We need to rethink this - I don't like the API definition here. Basically this is the `store` API with the only difference being that it depends on `metadata.publish_time` to set it as published. So, why do we need a new API for that? You could have just used the regular `store` API for that and maybe add a query param `publish: bool = False` to it.
My thinking is that if you create a new API, then you shouldn't provide the object again as payload - it would take the existing fset and modify it to set its `publish_time` and use that as a new version of the object (internally doing `store`). It should however get the new tag as query parameter, since the `reference` points to an existing object and now we need to pass both the `reference` and the new tag to assign to the new version.